### ISSUE TITLE
False reports of removed methods when extending class with generic parameters

### ISSUE THREAD

**Issue by batterseapower** [lvc/japi-compliance-checker#15]
If we have this before:

```
class A {
  void foo();
}
```

And then the new version has:

```
class A extends B<String> {}
class B<T> {
  void foo();
}
```

Then the compliance checker falsely reports that foo has been removed. I think this might be because it is trying to find the methods of a class called B<String> rather than simply "B" so it gets back an empty set.

A similar bug affects this case:

```
interface I {}
class A implements I {}
```

When it is changed to:

```
interface I {}
class A extends B<String> {}
class B<T> implements I {}
```

**Comment by lvc** [comment]
Hi,

This is fixed by the commit 9d78fb028752b905326eb67f9748ff1646102746 in master. It will be a part of the release `2.2`.

Please verify it.

Thank you.

**Comment by batterseapower** [comment]
Thank you for the fix!
